tss-esapi/src/abstraction/nv.rs

```diff
@@ -83,3 +78,185 @@ pub fn list(context: &mut Context) -> Result<Vec<(NvPublic, Name)>> {
         })
     })
 }
+
+/// Options and flags which can be used to determine how a non-volatile storage index is opened.
+///
+/// This builder exposes the ability to determine how a [`NvReaderWriter`] is opened, and is typically used by
+/// calling [`NvOpenOptions::new`], chaining method calls to set each option and then calling [`NvOpenOptions::open`].
+#[derive(Debug, Clone, Default)]
+pub struct NvOpenOptions {
+    nv_public: Option<NvPublic>,
+}
+
+impl NvOpenOptions {
+    /// Creates a new blank set of options for opening a non-volatile storage index
+    ///
+    /// All options are initially set to `false`/`None`.
+    pub fn new() -> Self {
+        Self { nv_public: None }
+    }
+
+    /// Sets the public attributes to use when creating the non-volatile storage index
+    ///
+    /// If the public attributes are `None` then the non-volatile storage index will be opened or otherwise
+    /// it will be created.
+    pub fn with_nv_public(&mut self, nv_public: Option<NvPublic>) -> &mut Self {
```

Comment:
Would it be possible to make a constructor out of this and then add a ```set_nv_public``` method in order to modify it later?

```rust
pub const fn with_nv_public(nv_public: NvPublic) -> Self {
    Self {nv_public: Some(nv_public) }
}
```

Reply:
Will redesign the whole thing in an upcoming PR.